app/src/main/java/com/duckduckgo/app/browser/BrowserViewModel.kt

```diff
@@ -388,6 +392,27 @@ class BrowserViewModel(
         viewState.value = currentViewState().copy(findInPage = findInPage)
     }
 
+
+    fun desktopSiteModeToggled(urlString: String?, desktopSiteRequested: Boolean) {
+        if (desktopSiteRequested) {
+            command.value = BrowserViewModel.Command.DesktopMode
+        } else {
+            command.value = BrowserViewModel.Command.MobileMode
```

Comment:
Are we sharing too much logic with the activity? Perhaps the command should be to update the user agent?

Reply:
As per the video chat conversation earlier, this might fall under any work that comes out of considering the wider command/view state approach.

Are you ok with deferring any more changes on this until the above has happened, @subsymbolic ?